File: spec/python/test_fixed_struct.py
import unittest

from fixed_struct import FixedStruct

class TestFixedStruct(unittest.TestCase):
    def test_fixed_struct(self):
        r = FixedStruct.from_file('src/fixed_struct.bin')

        self.assertEqual(r.header.uint8, 255)
        self.assertEqual(r.header.uint16, 65535)
        self.assertEqual(r.header.uint32, 4294967295)
        self.assertEqual(r.header.uint64, 18446744073709551615)

        self.assertEqual(r.header.sint8, -1)
        self.assertEqual(r.header.sint16, -1)
        self.assertEqual(r.header.sint32, -1)
        self.assertEqual(r.header.sint64, -1)

        self.assertEqual(r.header.uint16le, 66)
        self.assertEqual(r.header.uint32le, 66)
        self.assertEqual(r.header.uint64le, 66)

        self.assertEqual(r.header.sint16le, -66)
        self.assertEqual(r.header.sint32le, -66)
        self.assertEqual(r.header.sint64le, -66)

        self.assertEqual(r.header.uint16be, 66)
        self.assertEqual(r.header.uint32be, 66)
        self.assertEqual(r.header.uint64be, 66)

        self.assertEqual(r.header.sint16be, -66)
        self.assertEqual(r.header.sint32be, -66)
        self.assertEqual(r.header.sint64be, -66)

        r.close()

File: spec/javascript/test_fixed_struct.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('FixedStruct', 'src/fixed_struct.bin', function(r) {
  assert.equal(r.hdr.uint8, 255);
  assert.equal(r.hdr.uint16, 65535);
  assert.equal(r.hdr.uint32, 4294967295);
  assert.equal(r.hdr.uint64, 18446744073709551615);

  assert.equal(r.hdr.sint8, -1);
  assert.equal(r.hdr.sint16, -1);
  assert.equal(r.hdr.sint32, -1);
  assert.equal(r.hdr.sint64, -1);

  assert.equal(r.hdr.uint16le, 66);
  assert.equal(r.hdr.uint32le, 66);
  assert.equal(r.hdr.uint64le, 66);

  assert.equal(r.hdr.sint16le, -66);
  assert.equal(r.hdr.sint32le, -66);
  assert.equal(r.hdr.sint64le, -66);

  assert.equal(r.hdr.uint16be, 66);
  assert.equal(r.hdr.uint32be, 66);
  assert.equal(r.hdr.uint64be, 66);

  assert.equal(r.hdr.sint16be, -66);
  assert.equal(r.hdr.sint32be, -66);
  assert.equal(r.hdr.sint64be, -66);
});
